js/gantt-helper.js

```diff
@@ -0,0 +1,872 @@
+                           var dateToStr = gantt.date.date_to_str("%d %M");
+                           var endDate = gantt.date.add(date, 6, "day");
+                           var weekNum = gantt.date.date_to_str("%W")(date);
+                           return "#" + weekNum + ", " + dateToStr(date) + " - " + dateToStr(endDate);
+                        }
+                     },
+                     { unit: "day", step: 1, format: "%j %D" }
+                  ]
+               },
+               {
+                  name: "month",
+                  scale_height: 50,
+                  min_column_width: 120,
+                  scales: [{
+                        unit: "month",
+                        format: "%F, %Y"
+                     },
+                     {
+                        unit: "week",
+                        format: "Week #%W"
+                     }
+                  ]
+               },
+               {
+                  name: "quarter",
+                  height: 50,
+                  min_column_width: 90,
+                  scales: [
+                     { unit: "month", step: 1, format: "%M" },
+                     {
+                        unit: "quarter",
+                        step: 1,
+                        format: function(date) {
+                           var dateToStr = gantt.date.date_to_str("%M");
+                           var endDate = gantt.date.add(gantt.date.add(date, 3, "month"), -1, "day");
+                           return dateToStr(date) + " - " + dateToStr(endDate);
+                        }
+                     }
+                  ]
+               },
+               {
+                  name: "year",
+                  scale_height: 50,
+                  min_column_width: 30,
+                  scales: [
+                     { unit: "year", step: 1, format: "%Y" }
+                  ]
+               }
+            ]
+         };
+
+         gantt.ext.zoom.init(zoomConfig);
+         gantt.ext.zoom.setLevel("month");
+
+         // <<<<< Configs
+
+
+         // >>>>> Event handlers
+
+         gantt.ext.zoom.attachEvent("onAfterZoom", function(level, config) {
+            document.querySelector(".gantt_radio[value='" + config.name + "']").checked = true;
+         });
+
+         $("input[name='scale']").on('click', function(event) {
+            gantt.ext.zoom.setLevel(event.target.value);
+         });
+
+         $("input[name='branch_level']").on('click', function(event) {
+            expandCollapse(event.target.value);
+         });
+
+         $("input[name='branch_state']").on('click', function() {
+            expandCollapse($("input[name='branch_level']:checked").val());
+         });
+
+         gantt.attachEvent("onBeforeRowDragMove", function(id) {
+            $('#hf_gantt_item_state').val(id + "|" + gantt.getTask(id).parent);
+         });
+
+         gantt.attachEvent("onBeforeRowDragEnd", function(id, target) {
```

Comment:
1. An error appears when trying to move a project to the root level. Doing this should be possible.

2. An error appears when trying to move a task to the root level. This action should not be possible, so it should just be ignored and no error should be shown.